### PR TITLE
Bumping JSON.NET to 9.0.1

### PR DESCRIPTION
**Pull request by sokket**
Bumping our Json.NET version to 9.0.1-beta1 

Fixing bug #3377

/cc: @piotrpMSFT 

/fyi: @Eilon @javiercn

### CODE REVIEW

File: test/Microsoft.DotNet.ProjectModel.Tests/GivenThatIWantToLoadAProjectJsonFile.cs [hunk 1]
```diff
@@ -747,7 +747,7 @@ public void It_adds_the_dependency_when_the_version_is_set_directly()
             dependency.Type.Should().Be(LibraryDependencyType.Default);
             dependency.SourceFilePath.Should().Be(ProjectFilePath);
             dependency.SourceLine.Should().Be(3);
-            dependency.SourceColumn.Should().Be(31);
+            dependency.SourceColumn.Should().Be(30);
         }
```
Comment (piotrpMSFT) [inline test/Microsoft.DotNet.ProjectModel.Tests/GivenThatIWantToLoadAProjectJsonFile.cs]: Why did this value change?

File: test/Microsoft.DotNet.ProjectModel.Tests/GivenThatIWantToLoadAProjectJsonFile.cs [hunk 2]
```diff
@@ -769,7 +769,7 @@ public void It_adds_the_dependency_when_the_version_is_set_in_an_object()
             dependency.Type.Should().Be(LibraryDependencyType.Default);
             dependency.SourceFilePath.Should().Be(ProjectFilePath);
             dependency.SourceLine.Should().Be(3);
-            dependency.SourceColumn.Should().Be(25);
+            dependency.SourceColumn.Should().Be(24);
         }
```
Comment (piotrpMSFT) [inline test/Microsoft.DotNet.ProjectModel.Tests/GivenThatIWantToLoadAProjectJsonFile.cs]: Why did this value change?

File: test/Microsoft.DotNet.ProjectModel.Tests/GivenThatIWantToLoadAProjectJsonFile.cs [hunk 3]
```diff
@@ -909,7 +909,7 @@ public void It_adds_the_tool_when_the_version_is_set_directly()
             tool.Type.Should().Be(LibraryDependencyType.Default);
             tool.SourceFilePath.Should().Be(ProjectFilePath);
             tool.SourceLine.Should().Be(3);
-            tool.SourceColumn.Should().Be(25);
+            tool.SourceColumn.Should().Be(24);
         }
```
Comment (piotrpMSFT) [inline test/Microsoft.DotNet.ProjectModel.Tests/GivenThatIWantToLoadAProjectJsonFile.cs]: Why did this value change?

File: test/Microsoft.DotNet.ProjectModel.Tests/GivenThatIWantToLoadAProjectJsonFile.cs [hunk 4]
```diff
@@ -931,7 +931,7 @@ public void It_adds_the_tool_when_the_version_is_set_in_an_object()
             tool.Type.Should().Be(LibraryDependencyType.Default);
             tool.SourceFilePath.Should().Be(ProjectFilePath);
             tool.SourceLine.Should().Be(3);
-            tool.SourceColumn.Should().Be(19);
+            tool.SourceColumn.Should().Be(18);
```
Comment (piotrpMSFT) [inline test/Microsoft.DotNet.ProjectModel.Tests/GivenThatIWantToLoadAProjectJsonFile.cs]: Why did this value change?

File: test/Microsoft.DotNet.ProjectModel.Tests/GivenThatIWantToLoadAProjectJsonFile.cs (hunk 1 above)
Comment (sokket) [inline test/Microsoft.DotNet.ProjectModel.Tests/GivenThatIWantToLoadAProjectJsonFile.cs]: @javiercn why did these values change in your PR? I followed yours for the fixes to make this test work. What led you to change these values?

Comment (javiercn) [inline test/Microsoft.DotNet.ProjectModel.Tests/GivenThatIWantToLoadAProjectJsonFile.cs]: It's part of a bug fix that happened in Json.NET, I debugged through the code and we grab that value directly from Json.NET. The new position was just one space away or one " away.

File: test/Microsoft.DotNet.ProjectModel.Tests/GivenThatIWantToLoadAProjectJsonFile.cs (hunk 2 above)
Comment (javiercn) [inline test/Microsoft.DotNet.ProjectModel.Tests/GivenThatIWantToLoadAProjectJsonFile.cs]: Same as above

File: test/Microsoft.DotNet.ProjectModel.Tests/GivenThatIWantToLoadAProjectJsonFile.cs (hunk 3 above)
Comment (javiercn) [inline test/Microsoft.DotNet.ProjectModel.Tests/GivenThatIWantToLoadAProjectJsonFile.cs]: Same as above

File: test/Microsoft.DotNet.ProjectModel.Tests/GivenThatIWantToLoadAProjectJsonFile.cs (hunk 4 above)
Comment (javiercn) [inline test/Microsoft.DotNet.ProjectModel.Tests/GivenThatIWantToLoadAProjectJsonFile.cs]: Same as above

**Comment by eerhardt** [comment]
What is the difference between this PR and https://github.com/dotnet/cli/pull/3386?  Can we close one of them?

**Comment by piotrpMSFT** [comment]
@eerhardt it's due to a misscommunication. @Eilon asked us to fix this quickly in #3377 and we were not aware that #3386 was in progress since it was not lined to the issue. Since this PR does not have merge conflicts and both have failing builds I'm going to close #3386.

**Comment by piotrpMSFT** [comment]
@dotnet-bot test CentOS7.1 x64 Debug Build please.

**Comment by piotrpMSFT** [comment]
@dotnet-bot test Ubuntu16.04 x64 Debug Build please.

**Comment by piotrpMSFT** [comment]
#dotnet-bot test OSX x64 Release Build Please.

**Comment by piotrpMSFT** [comment]
@dotnet-bot test CentOS7.1 x64 Debug Build please.

**Comment by piotrpMSFT** [comment]
CentOS was out of disk space. Deleted the node and waiting for new one to get provisioned.

**Comment by sokket** [comment]
@dotnet-bot test  Debian8.2 x64 Debug Build please

**Comment by sokket** [comment]
@dotnet-bot test Ubuntu x64 Release Build please

**Comment by sokket** [comment]
@dotnet-bot test Windows_NT x64 Release Build  please

**Comment by sokket** [comment]
@dotnet-bot test Windows_NT x86 Debug Build please

**Comment by sokket** [comment]
Checks all pass, any further comments?

**Comment by sokket** [comment]
Resolving merge conflicts

**Comment by sokket** [comment]
@piotrpMSFT any other comments?

**Comment by piotrpMSFT** [comment]
:shipit:

**Pull request merged**